libraries/aws_security_group.rb

```diff
@@ -12,48 +12,53 @@ class AwsSecurityGroup < AwsResourceBase
     it { should exist }
   end
   "
-  attr_reader :description, :group_id, :group_name, :vpc_id, :inbound_rules, :outbound_rules, :inbound_rules_count, :outbound_rules_count, :tags
+  attr_reader :description, :group_id, :group_name, :vpc_id, :inbound_rules, :outbound_rules, :inbound_rules_count, :outbound_rules_count, :tags, :resource_data
 
   def initialize(opts = {})
     opts = { group_id: opts } if opts.is_a?(String)
     opts[:group_id] = opts.delete(:id) if opts.key?(:id) # id is an alias for group_id
 
     super(opts)
-    validate_parameters(require_any_of: %i(group_id group_name vpc_id))
 
-    filter = []
-    if opts.key?(:vpc_id)
```

Comment:
why are these removed?

Reply:
Its not removed .Since these lines were moved below It is showing removed.